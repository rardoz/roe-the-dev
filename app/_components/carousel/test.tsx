import Carousel from '.'
import { render } from '@testing-library/react'

describe('Carousel', () => {
  it('should render', () => {
    expect(
      render(
        <Carousel>
          <div>TEST</div>
        </Carousel>,
      ).container,
    ).toMatchSnapshot()
  })

  it('should have title set to Test 123', () => {
    expect(
      render(
        <Carousel>
          <div>Test 123</div>
        </Carousel>,
      ).container.textContent,
    ).toBe('Test 123')
  })
})
