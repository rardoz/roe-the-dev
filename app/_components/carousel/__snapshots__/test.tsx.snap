// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should render 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="relative h-full w-full"
      data-testid="carousel"
    >
      <div
        class="flex h-full snap-mandatory overflow-y-hidden overflow-x-scroll scroll-smooth rounded-none snap-x indiana-scroll-container indiana-scroll-container--hide-scrollbars"
      >
        <div
          class="w-full flex-shrink-0 transform cursor-grab snap-center"
          data-active="true"
          data-testid="carousel-item"
        >
          <div
            class="absolute top-1/2 left-1/2 block w-full -translate-x-1/2 -translate-y-1/2"
          >
            TEST
          </div>
        </div>
      </div>
      <div
        class="absolute bottom-5 left-1/2 flex -translate-x-1/2 space-x-3"
      >
        <button
          aria-label="Slide 1"
          class="h-3 w-3 rounded-full bg-white dark:bg-gray-800"
          data-testid="carousel-indicator"
        />
      </div>
      <div
        class="absolute top-0 left-0 flex h-full items-center justify-center px-4 focus:outline-none"
      >
        <button
          aria-label="Previous slide"
          class="group"
          data-testid="carousel-left-control"
          type="button"
        >
          <span
            class="inline-flex h-8 w-8 items-center justify-center rounded-full bg-white/30 group-hover:bg-white/50 group-focus:outline-none group-focus:ring-4 group-focus:ring-white dark:bg-gray-800/30 dark:group-hover:bg-gray-800/60 dark:group-focus:ring-gray-800/70 sm:h-10 sm:w-10 hidden sm:inline-flex"
          >
            <svg
              aria-hidden="true"
              class="h-5 w-5 text-white dark:text-gray-800 sm:h-6 sm:w-6"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 19l-7-7 7-7"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        class="absolute top-0 right-0 flex h-full items-center justify-center px-4 focus:outline-none"
      >
        <button
          aria-label="Next slide"
          class="group"
          data-testid="carousel-right-control"
          type="button"
        >
          <span
            class="inline-flex h-8 w-8 items-center justify-center rounded-full bg-white/30 group-hover:bg-white/50 group-focus:outline-none group-focus:ring-4 group-focus:ring-white dark:bg-gray-800/30 dark:group-hover:bg-gray-800/60 dark:group-focus:ring-gray-800/70 sm:h-10 sm:w-10 hidden sm:inline-flex"
          >
            <svg
              aria-hidden="true"
              class="h-5 w-5 text-white dark:text-gray-800 sm:h-6 sm:w-6"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 5l7 7-7 7"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
